internal/extension/smartagentextension/config.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package smartagentextension
+
+import (
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+type Config struct {
+	configmodels.ExtensionSettings `mapstructure:",squash"`
+	CollectdConfig                 CollectdConfig `mapstructure:"collectd"`
+}
+
+// Fields on this struct have a direct mapping to fields on
+// collect.CollectdConfig in the SignalFx Agent.
+// https://github.com/signalfx/signalfx-agent/blob/v5.7.2/pkg/core/config/config.go#L342
+type CollectdConfig struct {
+	// How many read intervals before abandoning a metric. Doesn't affect much
+	// in normal usage.
+	// See [Timeout](https://collectd.org/documentation/manpages/collectd.conf.5.shtml#timeout_iterations).
+	Timeout int `mapstructure:"timeout"`
+	// Number of threads dedicated to executing read callbacks. See
+	// [ReadThreads](https://collectd.org/documentation/manpages/collectd.conf.5.shtml#readthreads_num)
+	ReadThreads int `mapstructure:"read_threads"`
```

Comment:
The agent receiver supports the camel case struct tags from the agent to help w/ migrations.  Requiring the user to transform them here and not in the receiver seems a little counterintuitive to me.  Consistency is nice, but not entirely possible in this case, so inheriting the agent values is my preference.

Reply:
Updated the config to use camel case.